### PR TITLE
Us31 opti post form

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by LoicBrassart** [comment]
Please stop asking for my reviews too soon

**Review by Gumatan — DISMISSED** [review]

File: backend/index.js [hunk 1]
```diff
@@ -42,16 +44,51 @@ app.get("/api/posts/:limit", (req, res) => {
...
+        res.status(200).json(results);
+      }
+    }
+  );
+});
+
+app.post("/api/postimg", upload.single('file'), (req, res) => {
+
+  const formData = req.file;
+
```
Comment (LoicBrassart) [inline backend/index.js]: Several empty lines -> Prettier seems not to be aptly configured on your laptop

File: backend/index.js [hunk 2]
```diff
@@ -42,16 +44,51 @@ app.get("/api/posts/:limit", (req, res) => {
...
+        res.send(result);
+      }
+    });
+});
+
 app.post("/api/posts", (req, res) => {
   const formData = req.body;
   db.query("INSERT INTO post SET ?", formData, (err, results) => {
     if (err) {
+      console.log("2" + err)
```
Comment (LoicBrassart) [inline backend/index.js]: No console.log()

File: frontend/src/components/PostField.jsx [hunk 3]
```diff
@@ -1,22 +1,64 @@
...
+
+  const onSubmit = e => {
+    e.preventDefault();
+    if (file) {
+      let imageUpload = new FormData();
+      imageUpload.append("file", file);
+      axios.post(`${backend}/api/postimg`, imageUpload).then(response => {
+        let image_url = response.data.public_id;
+        if (game_id) {
+          var postObject = { image_url, message, user_id, game_id };
```
Comment (LoicBrassart) [inline frontend/src/components/PostField.jsx]: After 4 months, `var`, really ? I'm hurt, man... :-(

File: frontend/src/components/PostField.jsx [hunk 4]
```diff
@@ -1,22 +1,64 @@
...
+        } else {
+          var postObject = { image_url, message, user_id };
+        }
+
+        axios
+          .post(`${backend}/api/posts`, postObject)
+          .then(() => document.location.reload());
```
Comment (LoicBrassart) [inline frontend/src/components/PostField.jsx]: Doesn't this reload the whole page ? Breaking the whole "SPA" concept ?

File: frontend/src/components/Postcard.jsx [hunk 5]
```diff
@@ -43,13 +46,20 @@ function Postcard(props) {
...
-                  src="https://via.placeholder.com/500x300"
-                />
-              </div>
+              {props.image_url ? (
+                <div className="mediaContainer">
+                  {/* section with the media*/}
+                  <CloudinaryContext cloudName="lanur">
+                    <Image publicId={props.image_url} className="postmedia" />
+                  </CloudinaryContext>
+                  {/* <img className="postmedia" src={props.image_url} /> */}
```
Comment (LoicBrassart) [inline frontend/src/components/Postcard.jsx]: Dead code

**Review by LoicBrassart — CHANGES_REQUESTED** [review]
The feature may be there, but code quality went down quite badly, please check these comments

**Review by Gumatan — APPROVED** [review]

**Review by LoicBrassart — APPROVED** [review]
Much better! :ok_hand:

**Pull request merged**